chainerx/__init__.py

```diff
@@ -13,6 +13,7 @@
 
 if _available:
     from numpy import dtype, bool_, int8, int16, int32, int64, uint8, float32, float64  # NOQA
+    all_dtypes = (bool_, int8, int16, int32, int64, uint8, float32, float64)
```

Comment:
Should we prefix it with an underscore, as it doesn't have to be public?

Reply:
I think this is useful if it's public.
`self.dtype not in chainerx.all_dtypes` is more robust than `self.dtype == numpy.float16`.

Reply:
Okay, that makes sense. Could you write docs for it?

Reply:
Discussed offline that this should be documented in a "dtype" section, which we don't yet have so I'll resolve this issue for now.